src/utils.cpp

```diff
@@ -42,12 +42,25 @@ std::string sysError() noexcept
     return strerror(errno);
 }
 
-Utils::EVPKeyPtr readPublicKey(const std::string& fileName)
+Utils::EVPKeyPtr readPublicKey(const std::string& src)
 {
-    FilePtr fp(fopen(fileName.c_str(), "rb"));
-    if (!fp)
-        throw JWTXX::Key::Error("Can't open key file '" + fileName + "'. " + sysError());
-    return Utils::EVPKeyPtr(PEM_read_PUBKEY(fp.get(), nullptr, nullptr, nullptr));
+    // src is file name
+    FilePtr fp(fopen(src.c_str(), "rb"));
+    if (fp)
+        return Utils::EVPKeyPtr(PEM_read_PUBKEY(fp.get(), nullptr, nullptr, nullptr));
+
+    // src is key data
+    if (src.find("-----BEGIN") != 0)
```

Comment:
Why do we need this check. Wont PEM_read_bio_PUBKEY will fail if its invalid public key file.

Reply:
@Iyer-VivekAnand agree. This check has been removed